nmigen/lib/roundrobin.py

```diff
@@ -0,0 +1,47 @@
+from .. import *
+
+
+__all__ = ["RoundRobin"]
+
+
+class RoundRobin(Elaboratable):
+    """Round-robin scheduler.
+
+    For a given set of requests, the round-robin scheduler will
+    grant one request.
+
+    Parameters
+    ----------
+    width : int
+        Number of requests.
+
+    Attributes
+    ----------
+    requests : Signal(width), in
+        Set of requests.
+    grant : Signal(range(width)), out
+        Number of the granted request.
+    """
+    def __init__(self, *, width):
+        if not isinstance(width, int) or width < 0:
+            raise ValueError("Width must be a positive integer, not {!r}"
```

Comment:
```suggestion
            raise ValueError("Width must be a non-negative integer, not {!r}"
```

Zero width should be explicitly allowed, this is useful for generic code.